server/src/utils/request.ts

```diff
@@ -0,0 +1,101 @@
+import axios from 'axios';
+import FormData from 'form-data';
+import logger from '@shared/Logger';
+
+export async function post(endpoint_url: string, data: FormData, extra: any) {
+    return await request({ 
+        method: 'post',
+        url: endpoint_url,
+        data: data,
+        headers: extra}, true);
+}
+
+export async function get(endpoint_url: string, req_qry: any, token: string) {
+    return await request({ 
+        method: 'get',
+        url: endpoint_url,
+        params: req_qry,
+        headers: {
+            'Authorization': `Bearer ${token}`
+        }}, true);
+}
+
+export async function request(config: any, retry_flag: boolean) {
+    var resp = null
+    try {
+        resp = await axios(config);
+    } catch (error: any) {
+        // DEVELOPER NOTES:
+        // here handle errors per errors.md
+        console.log('Error message: [', error.message, ']');
+        if (error.response) {
+            console.log("response code: " + error.response.status)
+            console.log("response text: " + error.response.data)
+            // DEVELOPER NOTES:
+            // check for retryable (e.g. 500 & fhir) errors and do retrying...
+            if (retry_flag && is_retryable(error)) {
+                console.log("Request failed and is retryable, entering retry process...")
+                var retry_resp = await do_retry(config)
+                if (retry_resp) {
+                    resp = retry_resp;
+                }
+            }
+            else {
+                resp = error.response
+            }
+        }
+        else if (error.request) {
+            // something went wrong on sender side, not retryable
+            // error.request is an instance of XMLHttpRequest in the browser and an instance of
+            // http.ClientRequest in node.js
+            console.log("error.request: " + error.request);
+        }
+        // dump axios config for diagnosis
+        console.log("config:")
+        console.log(error.config);
+    }
+    return resp    
+}
+
+function is_retryable(error: any) {

```

Comment:
```suggestion
function isRetryable(error: Error) {

```

Reply:
The convention for JS is to use camel case, not snake case. Kind of weird to have mixed cases throughout the code base.